libraries/AP_BattMonitor/AP_BattMonitor_Sum.cpp

```diff
@@ -86,23 +94,34 @@ AP_BattMonitor_Sum::read()
             temperature_sum += temperature;
             temperature_count++;
         }
+
+        float consumed_mah;
+        if (_mon.consumed_mah(consumed_mah, i)) {
+            consumed_mah_sum += consumed_mah;
+        }
+
+        float consumed_wh;
+        if (_mon.consumed_wh(consumed_wh, i)) {
+            consumed_wh_sum += consumed_wh;
+        }
     }
     const uint32_t tnow_us = AP_HAL::micros();
-    const uint32_t dt_us = tnow_us - _state.last_time_micros;
 
     if (voltage_count > 0) {
-        _state.voltage = voltage_sum / voltage_count;
-    }
-    if (current_count > 0) {
-        _state.current_amps = current_sum;
+        if (option_is_set(AP_BattMonitor_Params::Options::Minimum_Voltage)) {
+            _state.voltage = voltage_min;
+        } else {
+            _state.voltage = voltage_sum / voltage_count;
+        }
     }
+    _state.current_amps = current_sum;
+    _state.consumed_mah = consumed_mah_sum;
+    _state.consumed_wh = consumed_wh_sum;
```

Comment:
This is a change in behavior if there is no current readings. Before it would hold at the old value, now it will go to zero. I think it is better to hold at the old value, particularly for the consumed readings. `current_count` could become a bool `has_current`. I think its safe to assume you have the consumed readings if you have current.

Reply:
Good point. Changed.